amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseCleanupManager.java

```diff
@@ -82,6 +82,7 @@
      */
     public void start() {
         log.debug("Starting lease cleanup thread.");
+        isRunning = true;
```

Comment:
what if the scheduler fails? We need to set it as running only when all the three operations here succeed. If one of them partially succeed, next time we should still be able to execute the same set of operation.